### PR TITLE
Preliminary integration tests and refactoring

### LINKED ISSUE #23: Implement integration & unit tests
Verify proper ROS Master protocol conformance

* Test using ros command line tools
* Test using rosnode
* ROS Graph under test
  * Topics
  * Params
  * Services

### PR DESCRIPTION
**Pull request by NickZ**
Fixes #23

Currently runs tests for roscpp and rosnode clients. Refactors vapor to no longer use singleton modules.

Fixes the bugs in parameter server. removeParam and subscribeParam were not working.

### CODE REVIEW

File: src/core-api.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ exports.shutdown = async function (req, res) {
 
   await coreUtil.logTouch(callerPath, null, req.ip)
 
-  return xmlrpc.sendResult([
+  xmlrpc.sendResult([
```
[the remark this answers is not in the document]
Comment (NickZ) [reply src/core-api.js]: Done

File: src/config.js [hunk 2]
```diff
@@ -11,47 +11,52 @@ const sanitize = require('sanitize-filename')
...
-
-    nconf.argv().env({logicalSeparator: '.'}).file({
+    
+    nconf.argv({whitelist: this.whitelist}).file({
       file: 'config.json',
       dir: this.basePath,
       search: true,
       logicalSeparator: '.'
-    }).defaults(this.defaults)
+    }).env({logicalSeparator: '.', whitelist: ['ROS_MASTER_URI']})
```
Comment (sevenbitbyte) [inline src/config.js]: `this.defaults` appears to be totally disregarded

File: README.md [hunk 3]
```diff
@@ -110,6 +122,13 @@ To interactively develop run:
 yarn
 yarn watch
 ```
+# Tests
+
+To run tests, first run `catkin_make` in the `tests/native_client` folder, and then source the file `tests/native_client/devel/setup.bash`.
```
[the remark this answers is not in the document]
Comment (NickZ) [reply README.md]: Nope, that's not required.

**Comment by sevenbitbyte** [comment]
LGTM

**Pull request merged**